File: two-sum/two-sum.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        use std::collections::HashMap;
        
        let mut map = HashMap::new();
        
        for (i, num) in nums.iter().enumerate() {
            
            match map.get(num){
                Some(&num) => return vec![num, i as i32],
                None => map.insert(target-num, i as i32)
            };
        }
        vec![]
    }
}

File: two-sum/two-sum.py
class Solution:
    def twoSum(self, nums, target):
        
        
        hash_map = {}
        
        for i, num in enumerate(nums):
            diff = hash_map.get(target-num)
            if diff:
                return [i, diff[0]]
            else: hash_map[num] = (i, num)